src/Details.jl

```diff
@@ -0,0 +1,246 @@
+### A Pluto.jl notebook ###
+# v0.19.38
+
+using Markdown
+using InteractiveUtils
+
+# This Pluto notebook uses @bind for interactivity. When running this notebook outside of Pluto, the following 'mock version' of @bind gives bound variables a default value (instead of an error).
+macro bind(def, element)
+    quote
+        local iv = try Base.loaded_modules[Base.PkgId(Base.UUID("6e696c72-6542-2067-7265-42206c756150"), "AbstractPlutoDingetjes")].Bonds.initial_value catch; b -> missing; end
+        local el = $(esc(element))
+        global $(esc(def)) = Core.applicable(Base.get, el) ? Base.get(el) : iv(el)
+        el
+    end
+end
+
+# ╔═╡ 81f5b495-76c4-4c54-93ab-b49c5ecb810a
+# ╠═╡ skip_as_script = true
+#=╠═╡
+begin
+	import Pkg
+	Pkg.activate(Base.current_project(@__DIR__))
+	Pkg.instantiate()
+	md"**Project env active**"
+end
+  ╠═╡ =#
+
+# ╔═╡ b7b18a54-afd7-4467-83ed-cc4f07c321fb
+using HypertextLiteral
+
+# ╔═╡ b3732e34-d331-4dd2-b4fb-11b2f397d7c1
+# ╠═╡ skip_as_script = true
+#=╠═╡
+const testslider = html"<input>"
+  ╠═╡ =#
+
+# ╔═╡ 13e81634-3b72-4b1d-a89b-36d184698d21
+const details_css = @htl("""
+<style type="text/css">
+pluto-output details {
+	border: 1px solid var(--rule-color);
+	border-radius: 4px;
+	padding: 0.5em 0.5em 0;
+	margin-block-start: 0;
+	margin-block-end: var(--pluto-cell-spacing);
+}
+
+pluto-output details:first-child {
+	margin-block-start: 0;
+}
+
+pluto-output details:last-child {
+	margin-block-end: 0;
+}
+
+pluto-output details summary {
+	cursor: pointer;
+	font-weight: bold;
+	margin: -0.5em -0.5em 0;
+	padding: 0.5em;
+	font-family: var(--system-ui-font-stack);
+}
+
+pluto-output details[open] {
+	padding: 0.5em;
+}
+
+pluto-output details[open] summary {
+	border-bottom: 1px solid var(--rule-color);
+	margin-bottom: 0.5em;
+}
+</style>
+""")
+
+# ╔═╡ df840588-23bd-4b03-b5ab-ef273052d198
+const Iterable = Union{AbstractVector, Tuple, Base.Generator}
+
+# ╔═╡ 46521e2b-ea06-491a-9842-13dff7dc8299
+begin
+	const embed_detail = isdefined(Main, :PlutoRunner) && isdefined(Main.PlutoRunner, :embed_display) ? Main.PlutoRunner.embed_display : identity
+	
+	function details(summary::AbstractString, contents::Iterable; open::Bool=false)
+		@htl("""
+		$(details_css)
+		<details $(open ? (open=true,) : nothing)>
+			<summary>$(summary)</summary>
+			<div class="details-content">
+				$(Iterators.map(contents) do detail
+					detail isa AbstractString ? 
+						@htl("<p>$(detail)</p>") : 
+						embed_detail(detail)
```

Comment:
I decided to drop the extra plutoui-details element, and wrap an AbstractString inside `<p>`. It looks good to me now, but could you check whether this is still the look that you intended?

Reply:
I love how this change would really simplify the code (and CSS!), yet it strikes at a core question: "What are the default assumptions and guarantees of the `details` function?" I should have expanded briefly on my philosophy and assumptions for `details` at the start of this PR so it could quickly be checked for alignment with the rest of the project, a lesson I will carry forward with me.

## Gotta wrap 'em all?

One goal of my initial design was to ensure consistent spacing between the items of the `content` argument as if they were all `<p>` elements under the assumption that this would be the expected behavior in the simple case and that advanced users could easily format their content items ahead of time with other well known methods (`md_str`, `combine`, etc) if necessary. I'm in favor of maintaining this assumption, but in the end it's your call.

With this change currently, Details.jl actually contains a convenient example of behavior I'd consider undesirable.

| ![Recording of Details.jl dictionary toggle](https://github.com/JuliaPluto/PlutoUI.jl/assets/11425944/6611d25d-7618-4e2b-aee4-4e942d7a653c) |
|:---:|
| _Demo of how some objects embedded by Pluto can now display unexpected behavior._ |

To fully demonstrate the impact I whipped up a mock notebook to make A/B comparison straightforward.

| ![Recording of alternative wrapping behavior demo](https://github.com/JuliaPluto/PlutoUI.jl/assets/11425944/54be30cc-b567-4693-9379-162f8bc9b675) |
|:--:|
| _Demo of the differences between wrapping all details and only wrapping `AbstractString` details._ |

## I just wanted to use `<p>` ;-;

Assuming we do want to continue wrapping all elements, `<p>` elements are unfortunately not valid when nested. A `md_str` can easily put us in a situation where `@htl` is forced to normalize the emitted DOM structure to something valid, leaving empty `<p>` tags on the final output. Visually, I see no difference, but it feels a little sloppy. I'm happy to pivot to a `<div>` with a class if that's favorable to the custom `<plutoui-detail>` element.

| ![Recording of alternative tag wrapping demo](https://github.com/JuliaPluto/PlutoUI.jl/assets/11425944/a588a1f0-86ef-45c1-8497-4e44b985fe64) |
|:--:|
| _Demo of the differences between wrapping all details in `<plutoui-detail>` and wrapping all details in `<p>`._ |